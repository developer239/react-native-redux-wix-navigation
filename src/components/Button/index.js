import React from 'react'
import { Button, Text } from 'native-base'
import styled from 'styled-components/native'

const StyledButton = styled(Button)`
  margin-bottom: 15px;
`

const ButtonComponent = ({ children, ...props }) =>
  <StyledButton {...props}><Text>{children}</Text></StyledButton>

export default ButtonComponent
