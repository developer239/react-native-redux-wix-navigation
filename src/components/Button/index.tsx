import React, { FunctionComponent } from 'react'
import { NativeBase, Text } from 'native-base'
import { StyledButton } from './styled'

export const Button: FunctionComponent<NativeBase.Button> = ({
  children,
  ...props
}) => (
  <StyledButton {...props}>
    <Text>{children}</Text>
  </StyledButton>
)
