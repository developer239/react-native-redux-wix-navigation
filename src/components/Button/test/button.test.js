// __tests__/Intro-test.js
import React from 'react'
import renderer from 'react-test-renderer'
import Button from '../index'

describe('Button', () => {
  test('renders correctly', () => {
    const onPressHandler = jest.fn()
    const tree = renderer
      .create(<Button onPress={onPressHandler}>Some Text</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
