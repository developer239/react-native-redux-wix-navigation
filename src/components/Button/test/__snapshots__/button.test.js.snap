// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#007aff",
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 5,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "borderWidth": undefined,
      "elevation": 2,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "space-between",
      "marginBottom": 15,
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "shadowColor": undefined,
      "shadowOffset": undefined,
      "shadowOpacity": undefined,
      "shadowRadius": undefined,
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#fff",
        "fontFamily": "System",
        "fontSize": 16.5,
        "marginLeft": 0,
        "marginRight": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
    uppercase={false}
  >
    Some Text
  </Text>
</View>
`;
